standard/expressions.md

```diff
@@ -968,6 +968,16 @@ For a function member that includes a parameter array, if the function member is
   - for a fixed value parameter or a value parameter created by the expansion, an implicit conversion ([§10.2](conversions.md#102-implicit-conversions)) exists from the argument expression to the type of the corresponding parameter, or
   - for a `ref` or `out` parameter, the type of the argument expression is identical to the type of the corresponding parameter.
 
+In the case of a method group
+
+- If the group contains both instance and static members, instance members invoked without an instance receiver or context are not applicable.
+- If the group contains both instance and static members, static members invoked with an instance receiver are not applicable.
```

Comment:
Do these lines mean that if there are *only* instance members or *only* static members, the members that will fail later are still applicable? What's the reasoning behind taking that approach rather than saying they're not applicable anyway?

Reply:
The reasoning is: this way we get a better error message. (This is how the compiler works.) We could potentially simplify this. There might be complexities around extension methods, and we're all somewhat confused.

Reply:
Suggestion: keep it like this for now, but raise an issue to see whether we can convince ourselves that the result is the same (in terms of validity etc) - the compiler can still choose to give a better error message.

Reply:
We now think that it could simplify things a lot if we could:

- Refer to methods rather than members, so that we don't need to say "In the case of a method group"
- Remove references to "If the group contains both instance and static members" so that we can just have the context of "the expression we're considering (e.g. whether it has an instance receiver), the method we're considering, and the argument list" - not the rest of the method group.

Reply:
(But we're not sure if this changes extension method invocation. Neal is checking.)

Reply:
We *should* remove the condition here. Demo code from Neal:

```csharp
using System;

static class Extensions {
    public static void M(this A a) {
        Console.WriteLine("extension A::M");
    }
}

class A {

    static void M() {
        Console.WriteLine("A::M");
    }
    static void Main() {
        A a = new A();
        a.M();
    }
}
```

This picks the extension method, *so must have determined that the static method is not applicable*.

Reply:
We should also remove the final validation bullet from 11.7.8.2.